### PR TITLE
Add version menu to CoW Swap

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Adds a version menu in the footer of CoW Swap
- Can close the menu by:
  - Click on link item
  - Click outside version menu
  - Click on X icon

Screenshot 2024-07-03 at 16 55 01

https://github.com/cowprotocol/cowswap/assets/31534717/29606d95-8b83-4c64-97fd-393f89baae7b

https://github.com/cowprotocol/cowswap/assets/31534717/db66f367-0309-480c-9e78-f1a481d830f4

https://github.com/cowprotocol/cowswap/assets/31534717/ca7447a1-96c6-47b7-a0cb-0c5d8ab91a63

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/application/containers/App/menuConsts.tsx [hunk 1]
```diff
@@ -5,6 +5,7 @@ import AppziButton from 'legacy/components/AppziButton'
 import { FortuneWidget } from 'modules/fortune/containers/FortuneWidget'
 
 import { Routes } from 'common/constants/routes'
+import { Version } from '../../../../legacy/components/Version'
```
Comment (fairlighteth) [inline apps/cowswap-frontend/src/modules/application/containers/App/menuConsts.tsx:8]: Should we move this from legacy to a proper folder elsewhere? As it's not legacy more?

Comment (alfetopito) [reply apps/cowswap-frontend/src/modules/application/containers/App/menuConsts.tsx:8]: Maybe. But you can shorten the import a bit:

```suggestion
import { Version } from 'legacy/components/Version'
```

**Review by alfetopito — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/application/containers/App/menuConsts.tsx (hunk 1 above)
Comment (fairlighteth) [reply apps/cowswap-frontend/src/modules/application/containers/App/menuConsts.tsx]: Thanks 👍🏼

**Comment by fairlighteth** [comment]
Re-factored code to only output the dropdown in the DOM if it needs to be shown.

**Review by anxolin — APPROVED** [review]
Looks great, nice to have it back

**Comment by elena-zh** [comment]
Great, thank you!

**Pull request merged**